pkg/tracer/tracer.go

```diff
@@ -881,25 +879,37 @@ func (m *FlowFetcher) LookupAndDeleteMap(met *metrics.Metrics) map[ebpf.BpfFlowI
 		}
 		flowPayload := model.BpfFlowContent{BpfFlowMetrics: &ebpf.BpfFlowMetrics{}}
 		flowPayload.AccumulateBase(&baseMetrics)
+		flows[id] = flowPayload
+	}
 
-		// Fetch additional metrics; ids are without direction and interface
-		shorterID := id
-		shorterID.Direction = 0
-		shorterID.IfIndex = 0
-		if err := m.objects.AdditionalFlowMetrics.LookupAndDelete(&shorterID, &additionalMetrics); err != nil {
-			if !errors.Is(err, cilium.ErrKeyNotExist) {
-				log.WithError(err).WithField("flowId", id).Warnf("couldn't lookup/delete additional metrics entry")
-				met.Errors.WithErrorName("flow-fetcher", "CannotDeleteAdditionalMetric").Inc()
-			}
-		} else {
-			for i := range additionalMetrics {
-				flowPayload.AccumulateAdditional(&additionalMetrics[i])
-			}
+	// Reiterate on additional metrics
+	var additionalMetrics []ebpf.BpfAdditionalMetrics
+	ids = []ebpf.BpfFlowId{}
+	addtlIterator := m.objects.AdditionalFlowMetrics.Iterate()
+	for addtlIterator.Next(&id, &additionalMetrics) {
+		ids = append(ids, id)
+	}
+
+	countAdditional := 0
+	for _, id := range ids {
+		countAdditional++
+		if err := m.objects.AdditionalFlowMetrics.LookupAndDelete(&id, &additionalMetrics); err != nil {
+			log.WithError(err).WithField("flowId", id).Warnf("couldn't lookup/delete additional metrics entry")
```

Comment:
why the error handling here is different than the other map in fact this more identical to old code so it should be similar no ? specially the check for `i===0` ?